csi/node_server.go

```diff
@@ -34,6 +97,18 @@ func (ns *NodeServer) NodePublishVolume(ctx context.Context, req *csi.NodePublis
 
 	targetPath := req.GetTargetPath()
 
+	if frontend, ok := req.VolumeAttributes["frontend"]; ok && frontend == "iscsi" {
```

Comment:
If the volume exists before and being used without `frontend` field in the Kubernetes API request, this code path won't be triggered. We should check the volume property using Longhorn API directly here.